bundle-workflow/src/test_workflow/test_recorder/test_recorder.py

```diff
@@ -0,0 +1,123 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# The OpenSearch Contributors require contributions made to
+# this file be licensed under the Apache-2.0 license or a
+# compatible open source license.
+
+import logging
+import os
+import shutil
+from typing import Dict
+
+import yaml
+
+from test_workflow.test_recorder.test_recorder_builder import \
+    TestRecorderBuilder
+
+
+class Singleton(type):
+    """Metaclass."""
+    _instances: Dict[type, object] = {}
+
+    def __call__(cls, *args, **kwargs):
+        if cls not in cls._instances:
+            cls._instances[cls] = super(Singleton, cls).__call__(*args, **kwargs)
+        else:
+            logging.info("TestRecorder is a singleton class, retrieving existing instance")
+        return cls._instances[cls]
+
+    def get_instance(cls):
+        if cls._instances is not None:
+            return cls._instances[cls]
+
+
+class TestRecorder(object, metaclass=Singleton):
+    ACCEPTABLE_TEST_TYPES = ["integ-test", "bwc-test", "perf-test"]
```

Comment:
We want to make sure that only these 3 types are used since these will be uploaded to S3 by test-publisher.
Hence the check here: https://github.com/opensearch-project/opensearch-build/pull/587/files#diff-25ffa61203a5f29c529cca6f3f8214485cc1fd74ed8ec22d690f079f14377a43R44

Reply:
[Discussed Offline] The test recorder should not be responsible for what test types are created or not.  If a new on is added this code should not need to be changed.  We should allow for any test type and if this creates problems we can fix as needed.